### PR TITLE
Pin zstd-sys to `v2.0.9` in parquet

### LINKED ISSUE #5565: `parquet / Build wasm32 (pull_request)` CI check failing on main
**Describe the bug**
`parquet / Build wasm32 (pull_request)` CI check is failing on main , evenwi

**To Reproduce**
Make a PR with no code changes, for example https://github.com/apache/arrow-rs/pull/5564

It fails like https://github.com/apache/arrow-rs/actions/runs/8474125669/job/23219908414?pr=5564

```
Run cargo build -p parquet --target wasm32-unknown-unknown
  cargo build -p parquet --target wasm3[2](https://github.com/apache/arrow-rs/actions/runs/8474125669/job/23219908414?pr=5564#step:6:2)-unknown-unknown
  shell: sh -e {0}
  env:
    RUSTFLAGS: -C debuginfo=1
    RUST_BACKTRACE: 1
   Compiling autocfg v1.2.0
   Compiling libm v0.2.8
   Compiling libc v0.2.15[3](https://github.com/apache/arrow-rs/actions/runs/8474125669/job/23219908414?pr=5564#step:6:3)
...
Compiling arrow-cast v51.0.0 (/__w/arrow-rs/arrow-rs/arrow-cast)
The following warnings were emitted during compilation:

warning: zstd-sys@2.0.10+zstd.1.5.6: In file included from zstd/lib/common/zstd_common.c:18:
warning: zstd-sys@2.0.10+zstd.1.5.6: In file included from zstd/lib/common/zstd_internal.h:35:
warning: zstd-sys@2.0.10+zstd.1.5.6: zstd/lib/common/xxhash.h:3157:22: error: called object type 'int' is not a function or function pointer
warning: zstd-sys@2.0.10+zstd.1.5.6:     XXH_STATIC_ASSERT(sizeof(XXH32_canonical_t) == sizeof(XXH32_hash_t));
warning: zstd-sys@2.0.10+zstd.1.5.6:     ~~~~~~~~~~~~~~~~~^
warning: zstd-sys@2.0.10+zstd.1.5.6: zstd/lib/common/xxhash.h:3600:22: error: called object type 'int' is not a function or function pointer
warning: zstd-sys@2.0.10+zstd.1.5.6:     XXH_STATIC_ASSERT(sizeof(XXH[64](https://github.com/apache/arrow-rs/actions/runs/8474125669/job/23219908414?pr=5564#step:6:65)_canonical_t) == sizeof(XXH64_hash_t));
warning: zstd-sys@2.0.10+zstd.1.5.6:     ~~~~~~~~~~~~~~~~~^
warning: zstd-sys@2.0.10+zstd.1.5.6: 2 errors generated.
...
```

**Expected behavior**
CI checks should pass when no code is chanegd

**Additional context**
Reported by @XiangpengHao  on https://github.com/apache/arrow-rs/pull/5564

### PR DESCRIPTION
**Pull request by Jefffrey**
# Which issue does this PR close?

Closes #5565

# Rationale for this change
 

Using `2.0.10` causes parquet wasm build to fail. Can pin to `2.0.9` to have succeeding CI again, and maybe wait for upstream to fix issue with higher versions.

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: parquet/Cargo.toml [hunk 1]
```diff
@@ -53,6 +53,7 @@ brotli = { version = "3.3", default-features = false, features = ["std"], option
 flate2 = { version = "1.0", default-features = false, features = ["rust_backend"], optional = true }
 lz4_flex = { version = "0.11", default-features = false, features = ["std", "frame"], optional = true }
 zstd = { version = "0.13.0", optional = true, default-features = false }
+zstd-sys = { version = "=2.0.9", optional = true, default-features = false }
```
Comment (kylebarron) [inline parquet/Cargo.toml:56]: Maybe you should make a new feature `zstd` that turns on both the `zstd` and the `zstd-sys` features? Should this be a range for `>=2.0,<2.10`?

Comment (Jefffrey) [reply parquet/Cargo.toml:56]: Yeah that's a good point :+1: 

I was just doing some testing here so I'll clean it up as suggested

**Review by tustvold — APPROVED** [review]
Is there an upstream ticket for this we can link to?

**Comment by Jefffrey** [comment]
> Is there an upstream ticket for this we can link to?

Updated in the code :+1:

**Pull request merged**